app/apps/route.js

```diff
@@ -3,6 +3,11 @@ import Ember from 'ember';
 export default Ember.Route.extend({
   model: function(){
     var stack = this.modelFor('stack');
-    return stack.get('apps');
+    var apps = stack.get('apps');
+    if (apps.get('isFulfilled')) {
```

Comment:
@sandersonet you are correct! `apps` is a promise proxy, and here we are inspecting the state of the promise to decide if it needs reloading or is already in flight.